install_requirements.py

```diff
@@ -0,0 +1,153 @@
+import os
+import platform
+import re
+import sys
+import subprocess
+
+# Before doing anything, cd to the directory containing this script.
+os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+# Find the names of the python tools to use.
+PYTHON_EXECUTABLE = os.getenv('PYTHON_EXECUTABLE')
+
+if not PYTHON_EXECUTABLE:
+    CONDA_DEFAULT_ENV = os.getenv('CONDA_DEFAULT_ENV')
+    if not CONDA_DEFAULT_ENV or CONDA_DEFAULT_ENV == "base" or not subprocess.call(["which", "python"]):
+        PYTHON_EXECUTABLE = "python3"
+    else:
+        PYTHON_EXECUTABLE = "python"
+
+if PYTHON_EXECUTABLE == "python":
+    PIP_EXECUTABLE = "pip"
+else:
+    PIP_EXECUTABLE = "pip3"
+
+print(f"Using Python executable: {PYTHON_EXECUTABLE}")
+print(f"Using Pip executable: {PIP_EXECUTABLE}")
+
+def python_is_compatible():
+    # Scrape the version range from pyproject.toml, which should be in the current directory.
+    version_specifier = None
+    with open('pyproject.toml', 'r') as file:
+        for line in file:
+            if line.startswith('requires-python'):
+                match = re.search(r'"([^"]*)"', line)
+                if match:
+                    version_specifier = match.group(1)
+                    break
+
+    if not version_specifier:
+        print("WARNING: Skipping python version check: version range not found", file=sys.stderr)
+        return False
+
+    # Install the packaging module if necessary.
+    try:
+        import packaging
+    except ImportError:
+        subprocess.check_call([PIP_EXECUTABLE, 'install', 'packaging'])
```

Comment:
Throughout this file, please use `supprocess.run(..., check=True)` [instead](https://docs.python.org/3/library/subprocess.html#subprocess.run). This is the more modern API.